tutor/src/screens/assignment-grade/question.js

```diff
@@ -93,31 +92,78 @@ const ExpandGraded = observer(({ ux }) => {
 
 
 const AssignmentGradingQuestion = observer(({ ux }) => (
-  <StyledQuestion>
-    <QuestionHeader ux={ux} />
-    <QuestionBody>
-      <Question
-        question={ux.selectedHeading ? ux.selectedHeading.question : null}
-        hideAnswers={false}
-        displaySolution={ux.showAnswerKey}
-        choicesEnabled={false}
-        displayFormats={false}
-      />
-      <ExpandGraded ux={ux} />
-      {ux.visibleResponses.map((response, index) =>
-        <Answer
-          response={response}
-          key={index} index={index}
-          ux={ux} />)}
-    </QuestionBody>
-  </StyledQuestion>
+  <>
+    {
+      ux.showOverview
+        ? (
+          <>
+            {
+              ux.headings.map((h, i) => (
+                <StyledQuestion key={i} marginBottom>
+                  <QuestionHeader questionIndex={i} ux={ux} />
+                  <QuestionBody>
+                    <Question
+                      question={h.question}
+                      hideAnswers={false}
+                      displaySolution={false}
+                      choicesEnabled={false}
+                      displayFormats={false}
+                    />
+                    <ExpandGradedWrapper>
+                      <Button
+                        onClick={() => ux.goToQuestionHeading(i, true)}
+                        variant="link"
+                      >
+                        Expand graded answers {h.gradedProgress}
+                      </Button>
+                      <label>
+                        Average Score: {S.numberWithOneDecimalPlace(h.averageGradedPoints)} out of {S.numberWithOneDecimalPlace(h.responseStats.availablePoints)}
+                      </label>
+                    </ExpandGradedWrapper>
+                  </QuestionBody>
+                </StyledQuestion>
+              ))
+            }
+          </>
+        )
+        : (
+          <StyledQuestion>
+            <QuestionHeader questionIndex={ux.selectedHeading.index} ux={ux} showAnswerKey={true} />
+            <QuestionBody>
+              <Question
+                question={ux.selectedHeading ? ux.selectedHeading.question : null}
+                hideAnswers={false}
+                displaySolution={ux.showAnswerKey}
+                choicesEnabled={false}
+                displayFormats={false}
+              />
+              <ExpandGraded ux={ux} />
+              {
+                Boolean(ux.expandGradedAnswers) && ux.completedResponses.map((response, index) => 
+                  <Answer
+                    response={response}
+                    key={index}
+                    index={index}
+                    ux={ux} />)
+              }
+              {ux.needsGradingResponses.map((response, index) =>
+                <Answer
+                  response={response}
+                  key={index}
+                  index={index}
+                  ux={ux} />)}
+            </QuestionBody>
+          </StyledQuestion>
+        )
+    }
+  </>
 ));
```

Comment:
hm, instead of a huge ternary, how about extracting the two sides into separate components and:
```
const AssignmentGradingQuestion = observer(({ ux }) => (
  const Component = ux.showOverview ? Overview : Question;
 return <Component ux={ux} />;
})
```

Reply:
that is a good idea! :+1: